### ISSUE TITLE
LLM fail to instruction following

### ISSUE THREAD

**Issue by MSungK** [richard-peng-xia/MMed-RAG#22]
Hello, I am currently working on reproducing your excellent work.
I have trained an LLM and conducted evaluations based on the provided training dataset and code.
During this process, I observed a tendency for the model to fail to follow the instruction (i.e., not answering “yes” or “no”) as the number of retrieved reports increased.
Therefore, I would like to ask a few questions:
	1.	Did you conduct the experiments reported in the paper using the provided training dataset? - the provided training dataset which used in DPO only includes one report
	2.	Could you share the top_k value and retrieval threshold hyperparameters used for each benchmark?– I have noticed that adding just 2 to 3 reports often causes the model to fail in following the instruction.

**Comment by richard-peng-xia** [comment]
Hi @MSungK,

1. Yes, we do the DPO training as you said above. The training data we use only has one retrieval report.
2. I remember that the best k value should be between 6 and 10. For long-context reports, the model may have difficulty following the instructions, such as MIMIC-CXR.

**Comment by MSungK** [comment]
In that case, it seems that the model was tuned unstably during the DPO training process. Could you share the detailed hyperparameters used for the training?
“I conducted the training under the following conditions.”

deepspeed --include localhost:0,1,4,5 ../train/dpo/train_dpo_2stages.py \
    --model_name_or_path microsoft/llava-med-v1.5-mistral-7b \
    --deepspeed zero3.json \
    --version v1 \
    --lora_enable True --lora_r 128 --lora_alpha 256 --mm_projector_lr 2e-5 \
    --data_path /workspace/Minsung/MedSurvey/External/MMed-RAG/data/training/alignment/ophthalmology/harvard_vqa.json \
    --image_folder /workspace/Minsung/MedSurvey/External/MMed-RAG/data/alignment/harvard \
    --vision_tower openai/clip-vit-large-patch14-336 \
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio pad \
    --group_by_modality_length True \
    --bf16 True \
    --output_dir /workspace/Minsung/MedSurvey/External/MMed-RAG/logs/harvard \
    --num_train_epochs 3 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 200 \
    --save_total_limit 1 \
    --learning_rate 1e-7 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 1024 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True

**Comment by richard-peng-xia** [comment]
The script seems fine for me. Maybe you can check the data format or reduce the learning rate? Thanks.
